tools/smartdnsblock/smartdnsblock/smartdnsblock.cpp

```diff
@@ -35,15 +35,31 @@ ULONG GET_ADAPTERS_ADDRESSES_BUFFER_SIZE = 16384;
 PCWSTR FILTER_PROVIDER_NAME = L"Outline";
 PCWSTR SUBLAYER_NAME = L"Smart DNS Block";
 
+UINT64 MAX_TRIES = 3;
 UINT64 LOWER_FILTER_WEIGHT = 10;
 UINT64 HIGHER_FILTER_WEIGHT = 20;
 
 int main(int argc, char **argv) {
   // Lookup the interface index of outline-tap0.
-  PIP_ADAPTER_ADDRESSES adaptersAddresses =
-      (IP_ADAPTER_ADDRESSES *)malloc(GET_ADAPTERS_ADDRESSES_BUFFER_SIZE);
-  DWORD result = GetAdaptersAddresses(AF_INET, 0, NULL, adaptersAddresses,
-                                      &GET_ADAPTERS_ADDRESSES_BUFFER_SIZE);
+  PIP_ADAPTER_ADDRESSES adaptersAddresses = NULL;
```

Comment:
Could you add a comment with explaining that we retry in order to recover from `ERROR_BUFFER_OVERFLOW` per https://docs.microsoft.com/en-us/windows/win32/api/iphlpapi/nf-iphlpapi-getadaptersaddresses#return-value?

Reply:
Comments added.